editor/components/inserter/menu.js

```diff
@@ -148,180 +159,23 @@ export class InserterMenu extends Component {
-
-		// If the block list is empty return early.
-		if ( ! sortedByCategory.length ) {
-			return;
-		}
-
-		const nextBlock = this.findNext( sortedByCategory );
-		this.changeMenuSelection( nextBlock );
-	}
-
-	focusPrevious() {
-		const sortedByCategory = flow(
-			this.searchBlocks,
-			this.sortBlocks,
-		)( this.getBlocksForCurrentTab() );
-
-		// If the block list is empty return early.
-		if ( ! sortedByCategory.length ) {
-			return;
-		}
-
-		const nextBlock = this.findPrevious( sortedByCategory );
-		this.changeMenuSelection( nextBlock );
-	}
-
-	onKeyDown( keydown ) {
-		switch ( keydown.keyCode ) {
-			case TAB:
-				if ( keydown.shiftKey ) {
-					// Previous.
-					keydown.preventDefault();
-					this.focusPrevious( this );
-					break;
-				}
-				// Next.
-				keydown.preventDefault();
-				this.focusNext( this );
-				break;
-
-			case LEFT:
-				if ( this.state.currentFocus === 'search' ) {
-					return;
-				}
-				this.focusPrevious( this );
-				break;
-
-			case UP:
-				keydown.preventDefault();
-				this.focusPrevious( this );
-				break;
-
-			case RIGHT:
-				if ( this.state.currentFocus === 'search' ) {
-					return;
-				}
-				this.focusNext( this );
-				break;
-
-			case DOWN:
-				keydown.preventDefault();
-				this.focusNext( this );
-				break;
-
-			default:
-				return;
-		}
-
-		// Since unhandled key will return in the default case, we can assume
-		// having reached this point implies that the key is handled.
-		keydown.stopImmediatePropagation();
-	}
-
-	changeMenuSelection( refName ) {
-		this.setState( {
-			currentFocus: refName,
-		} );
-
-		// Focus the DOM node.
-		this.nodes[ refName ].focus();
-	}
-
-	setSearchFocus() {
-		this.changeMenuSelection( 'search' );
-	}
-
-	getBlockItem( block ) {
-		const disabled = this.isDisabledBlock( block );
-		return (
-			<button
-				role="menuitem"
-				key={ block.name }
-				className="editor-inserter__block"
-				onClick={ this.selectBlock( block.name ) }
-				ref={ this.bindReferenceNode( block.name ) }
-				tabIndex="-1"
-				onMouseEnter={ ! disabled ? this.props.showInsertionPoint : null }
-				onMouseLeave={ ! disabled ? this.props.hideInsertionPoint : null }
-				disabled={ disabled }
-			>
-				<BlockIcon icon={ block.icon } />
-				{ block.title }
-			</button>
-		);
-	}
-
-	renderBlocks( blocks, separatorSlug ) {
+	renderBlocks( blockTypes, separatorSlug ) {
 		const { instanceId } = this.props;
 		const labelledBy = separatorSlug === undefined ? null : `editor-inserter__separator-${ separatorSlug }-${ instanceId }`;
+		const blockTypesInfo = blockTypes.map( ( blockType ) => (
+			{ ...blockType, disabled: this.isDisabledBlock( blockType ) }
+		) );
 		return (
-			<div
-				className="editor-inserter__category-blocks"
-				tabIndex="0"
-				aria-labelledby={ labelledBy }
-			>
-				{ blocks.map( ( block ) => this.getBlockItem( block ) ) }
-			</div>
+			<InserterGroup blockTypes={ blockTypesInfo } labelledBy={ labelledBy }
```

Comment:
This convention for applying props is quite difficult to read. I'd suggest either keeping it on a single line, to a certain character limit (80-100), then otherwise consistently applying props one-per-line:

```jsx
<InserterGroup
	blockTypes={ blockTypesInfo }
	labelledBy={ labelledBy }
	bindReferenceNode={ this.bindReferenceNode }
	selectBlock={ this.selectBlock }
/>
```

Reply:
Also, we lost the `showInsertionPoint`, `hideInsertionPoint` props in this refactor. See #3613.